addons/web_tour/static/src/js/tip.js

```diff
@@ -110,19 +118,91 @@ var Tip = Widget.extend({
         return this._super.apply(this, arguments);
     },
     update: function ($anchor) {
-        if (!$anchor.is(this.$anchor)) {
-            this._unbind_anchor_events();
-            this.$anchor = $anchor;
-            this.$ideal_location = this._get_ideal_location();
-            if (this.$el.parent()[0] !== document.body) {
-                this.$el.appendTo(this.$ideal_location);
+        this._unbind_anchor_events();
+        this._setupAnchor($anchor);
+        this._updatePosition();
+        this._reposition();
+        this._bind_anchor_events();
+    },
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    _setupAnchor: function ($anchor) {
+        this.$anchor = $anchor;
+        this.$actualAnchor = this.$anchor;
+        this.$ideal_location = this._get_ideal_location();
+        this.$furtherIdealLocation = this._get_ideal_location(this.$ideal_location);
+    },
+    /**
+     * This function figure out which direction the tip should take
+     * and if he's at the bottom or the top of the targeted element
+     * or if it's an indicator to scroll.
+     */
+    _updatePosition: function () {
+        let moveTip = false;
+        let halfHeight = 0;
+        if (this.initialPosition === 'right' || this.initialPosition === 'left') {
+            halfHeight = this.$anchor.innerHeight() / 2;
+        }
+        let position = this.info.position;
+
+
+        const paddingTop = parseInt(this.$ideal_location.css('padding-top'));
+        let topViewport = window.pageYOffset;
+        let botViewport = topViewport + window.innerHeight;
+        topViewport += paddingTop;
+        let topOffset = this.$anchor.offset().top;
+        let botOffset = topOffset + this.$anchor.innerHeight();
+        /* Check if the viewport state change to know if we need to move the anchor of the tip.
+        * up : the target element is above the current viewport
+        * down : the target element is below the current viewport
+        * in : the target element is in the current viewport
+        */
```

Comment:
```suggestion
        // Check if the viewport state change to know if we need to move the anchor of the tip.
        // up : the target element is above the current viewport
        // down : the target element is below the current viewport
        // in : the target element is in the current viewport
```
To solve the alignment problem, it's driving me crazy.